go-test: add page (#6130)
tldr-pages/tldr: pages/common/go-test.md

# go test

> Tests Go packages (files have to end with `_test.go`).
> More information: <https://golang.org/cmd/go/#hdr-Testing_flags>.

- Test the package found in the current directory:

`go test`

- [v]erbosely test the package in the current directory:

`go test -v`

> - Test the packages in the current directory and all subdirectories:
> 
> `go test -v ./...`

Reviewer: This example is confusing a bit — is this example basically equivalent (and true)?
```suggestion
- Test the packages in the specified directory and all subdirectories:

`go test -v {{path/to/directory}}`
```

Author: The `go test` tool always takes into consideration the current directory, hence why in the first example no argument was required. The `./` part of the argument specifies that the package in the current directory must be tested, and the `...` part relates to all the other subdirectories, recursively.

Author: Clarifying, the given examples would not be equivalent. If you pass a specific directory path to the `go test` command, it'll only test the packages found at that directory, therefore ignoring any subdirectory.

Reviewer: Oh, I see. So `...` isn't just a placeholder value, you'd actually pass it to the command?

Author: Yes, that is correct.

Reviewer: Then maybe
```suggestion
- Test the packages in the current directory and all subdirectories (note the `...`):

`go test -v ./...`
```

- Test the package in the current directory and run all benchmarks:

> `go test -v -bench .`

Reviewer: ```suggestion
`go test -v -bench {{.}}`
```

> - [v]erbosely test the packages in the current directory and all subdirectories:
> 
> `go test -v ./...`
> 
> - [v]erbosely test the package in the current directory and run all benchmarks:
> 
> `go test -v -bench .`
> 
> - [v]erbosely test the package in the current directory and run all benchmarks for 50 seconds:

Reviewer: Since `-v` is in all of the examples, why not just omit explaining it?  I'd like to keep using `[` as the first character at a minimum.
```suggestion
- Test the packages in the current directory and all subdirectories:

`go test -v ./...`

- Test the package in the current directory and run all benchmarks:

`go test -v -bench .`

- Test the package in the current directory and run all benchmarks for 50 seconds:
```

> `go test -v -bench . -benchtime 50s`

Reviewer: ```suggestion
`go test -v -bench {{.}} -benchtime {{50s}}`
```

Author: I suppose this reply relates to the the previous comment/suggestion, but the `.` would actually be interpreted as a regular expression if changed. I never mentioned this in the examples as I assumed this would break the 5th guideline mentioned in the contributing guidelines, that being
> Focus on details specific to the command, and avoid explaining general UNIX concepts that could apply to any command (ex: relative/absolute paths, glob patterns/wildcards, special character escaping...).

So I just attempted to keep the examples simple in that regard.

If that really is the case, per my understanding the curly brackets wouldn't be necessary around the `.` character.

Reviewer: You're right, it's good to avoid introducing general UNIX concepts; however, regular expressions is really a general computer science concept, and use on lots and lots of other pages, so I think it's okay (and advisable) to clarify that here. 

>If that really is the case, per my understanding the curly brackets wouldn't be necessary around the `.` character.

We always put the token syntax (`{{` and `}}`) around user-mutable parameters/arguments.  If the `.` should always be a `.` to make the example correct, then let's leave it as-is.  Otherwise, if it can be changed without making the description inaccurate, put curly braces around it.  Assuming the former:
```suggestion
`go test -v -bench . -benchtime {{50s}}`
```
